src/vs/base/common/filters.ts

```diff
@@ -468,21 +468,7 @@ function isSeparatorAtPos(value: string, index: number): boolean {
 		return false;
 	}
 	const code = value.charCodeAt(index);
-	switch (code) {
-		case CharCode.Underline:
-		case CharCode.Dash:
-		case CharCode.Period:
-		case CharCode.Space:
-		case CharCode.Slash:
-		case CharCode.Backslash:
-		case CharCode.SingleQuote:
-		case CharCode.DoubleQuote:
-		case CharCode.Colon:
-		case CharCode.DollarSign:
-			return true;
-		default:
-			return false;
-	}
+	return isWordSeparator(code) || strings.isEmojiImprecise(code);
```

Comment:
I see how it is inconsistent of us having two ways to identify word separators but I would not make this change (removal of switch-case) here. For once the sets of characters are different and then I believe that using a set over a switch-case is slower

Reply:
Thanks for the comment! I'm still working on testing which symbol should be added as a separator. I saw some test cases failed and I haven't found the cause